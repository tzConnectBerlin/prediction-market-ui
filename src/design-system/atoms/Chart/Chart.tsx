import React from 'react';
import { ChartData, ChartOptions } from 'chart.js';
import { Bar, Line } from 'react-chartjs-2';

export interface ChartProps {
  chartType?: 'BarChart' | 'LineChart';
  data: ChartData;
  options: ChartOptions;
}

export const Chart: React.FC<ChartProps> = ({ chartType = 'LineChart', data, options }) => {
  const PlotLineChart = () => {
    return <Line type="line" data={data} options={options} />;
  };
  const PlotBarChart = () => {
    return <Bar type="bar" data={data} options={options} />;
  };

  return <>{chartType === 'LineChart' ? PlotLineChart() : PlotBarChart()}</>;
};
