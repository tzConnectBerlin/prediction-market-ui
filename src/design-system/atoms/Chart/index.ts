export { ChartContainer as default, ChartContainer } from './ChartContainer';
export type { ChartProps } from './Chart';
