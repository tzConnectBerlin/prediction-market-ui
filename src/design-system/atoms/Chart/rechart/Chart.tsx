import React from 'react';
import {
  Bar,
  BarChart,
  CartesianGrid,
  Legend,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
} from 'recharts';

export interface Axis {
  value: string;
  y_position?: 'left' | 'right';
  x_position?: 'top' | 'bottom';
}

export interface ChartProps {
  chartType?: 'BarChart' | 'LineChart';
  data: any[];
  colors: string[];
  yAxis: Axis[];
  xAxis: Axis;
}

export const Chart: React.FC<ChartProps> = ({
  chartType = 'LineChart',
  data,
  colors,
  yAxis,
  xAxis,
}) => {
  const PlotLineChart = () => {
    const lines = Object.keys(data[0]).slice(1);
    return (
      <LineChart
        width={500}
        height={300}
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="0" vertical={false} />
        <XAxis dataKey="name" axisLine={false} label={xAxis} />
        {yAxis.map((y) => (
          <YAxis
            key={y.value}
            yAxisId={y.y_position}
            axisLine={false}
            tickLine={false}
            orientation={y.y_position}
            label={{ ...y, angle: y.y_position === 'left' ? -90 : 90 }}
          />
        ))}
        <Tooltip />
        <Legend verticalAlign="top" align="left" height={50} />
        {lines.map((item, i) => (
          <Line
            type="monotone"
            yAxisId={yAxis[i].y_position}
            dataKey={item}
            stroke={colors[i]}
            key={i}
            dot={false}
          />
        ))}
      </LineChart>
    );
  };
  const PlotBarChart = () => {
    const lines = Object.keys(data[0]).slice(1);
    return (
      <BarChart
        width={500}
        height={300}
        data={data}
        margin={{
          top: 5,
          right: 30,
          left: 20,
          bottom: 5,
        }}
      >
        <CartesianGrid strokeDasharray="0" vertical={false} />
        <XAxis dataKey="name" axisLine={false} label={xAxis} />
        {yAxis.map((y) => (
          <YAxis
            key={y.value}
            yAxisId={y.y_position}
            axisLine={false}
            tickLine={false}
            orientation={y.y_position}
            label={{ ...y, angle: y.y_position === 'left' ? -90 : 90 }}
          />
        ))}
        <Tooltip />
        <Legend verticalAlign="top" align="left" height={50} />
        {lines.map((item, i) => (
          <Bar
            type="monotone"
            yAxisId={yAxis[i].y_position}
            dataKey={item}
            fill={colors[i]}
            key={i}
          />
        ))}
      </BarChart>
    );
  };

  return (
    <>
      <div style={{ width: '90vm', height: '250px' }}>
        <ResponsiveContainer width="100%" height="100%">
          {chartType === 'LineChart' ? PlotLineChart() : PlotBarChart()}
        </ResponsiveContainer>
      </div>
    </>
  );
};
