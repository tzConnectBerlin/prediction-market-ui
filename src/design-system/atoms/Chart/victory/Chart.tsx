import React from 'react';
import {
  VictoryBar,
  VictoryChart,
  VictoryAxis,
  VictoryTheme,
  VictoryStack,
  VictoryGroup,
  VictoryLegend,
} from 'victory';

export interface Axis {
  value: string;
  y_position?: 'left' | 'right';
  x_position?: 'top' | 'bottom';
}

export interface ChartProps {
  chartType?: 'BarChart' | 'LineChart';
  data: any[];
  colors: string[][];
  yAxis?: Axis[];
  xAxis?: Axis;
}

export const Chart: React.FC<ChartProps> = ({
  chartType = 'BarChart',
  data,
  colors,
  yAxis,
  xAxis,
}) => {
  const PlotLineChart = () => {
    const lines = Object.keys(data[0]).slice(1);
    return (
      <VictoryChart>
        <VictoryBar data={data} x="name" y="yes" />
        <VictoryBar data={data} x="name" y="no" />
      </VictoryChart>
    );
  };
  const PlotBarChart = () => {
    const lines = Object.keys(data[0]).slice(1);
    return (
      <VictoryChart>
        <VictoryLegend
          x={125}
          y={50}
          orientation="horizontal"
          gutter={20}
          style={{ title: { fontSize: 20 } }}
          data={[
            { name: 'Yes', symbol: { fill: colors[0][0] } },
            { name: 'No', symbol: { fill: colors[1][0] } },
          ]}
        />
        <VictoryGroup offset={20} style={{ data: { width: 15 } }}>
          {data.map((d, i) => {
            return (
              <VictoryStack colorScale={colors[i]} key={i}>
                <VictoryBar data={d} />
              </VictoryStack>
            );
          })}
        </VictoryGroup>
      </VictoryChart>
    );
  };

  return (
    <>
      <div>{chartType === 'LineChart' ? PlotLineChart() : PlotBarChart()}</div>
    </>
  );
};
