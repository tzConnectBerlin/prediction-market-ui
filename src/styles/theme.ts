import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
