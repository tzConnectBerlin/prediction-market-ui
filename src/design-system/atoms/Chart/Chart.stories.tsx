import { Story, Meta } from '@storybook/react/types-6-0';
import { Chart, ChartProps } from './Chart';
import { lightTheme as theme } from '../../../theme';

export default {
  title: 'Atom/Chart',
  component: Chart,
} as Meta;

const defaultChartData = {
  labels: ['3 MAR 2021', '4 MAR 2021', '5 MAR 2021', '6 MAR 2021', '7 MAR 2021'],
  datasets: [
    {
      label: 'Yes',
      data: [1, 2, 3, 2.5, 4],
      borderColor: theme.palette.success.main,
      backgroundColor: theme.palette.success.main,
      yAxisID: 'y',
    },
  ],
};

const newChartData = {
  label: 'No',
  data: [20, 35, 25, 15, 17],
  borderColor: theme.palette.error.main,
  backgroundColor: theme.palette.error.main,
  yAxisID: 'y1',
};

const chartMultiData = {
  ...defaultChartData,
  datasets: defaultChartData.datasets.concat(newChartData),
};

const chartOptions = {
  responsive: true,
  scales: {
    x: {
      grid: {
        display: false,
      },
    },
    y: {
      type: 'linear',
      display: true,
      position: 'left',
      title: { display: true, text: 'Yes' },
      grid: {
        borderColor: 'transparent',
      },
    },
    y1: {
      type: 'linear',
      display: true,
      position: 'right',
      title: { display: true, text: 'No' },
      grid: {
        borderColor: 'transparent',
        drawOnChartArea: false, // only want the grid lines for one axis to show up
      },
    },
  },
};

const Template: Story<ChartProps> = (args) => <Chart {...args} />;

export const Default = Template.bind({});
Default.args = {
  chartData: defaultChartData,
};

export const BarChart = Template.bind({});
BarChart.args = {
  chartType: 'bar',
  chartData: defaultChartData,
  chartTitle: 'Test Title',
};

export const MultiLineChart = Template.bind({});
MultiLineChart.args = {
  chartData: chartMultiData,
  options: chartOptions,
  chartTitle: 'Test Title',
};

export const MultiBarChart = Template.bind({});
MultiBarChart.args = {
  chartType: 'bar',
  chartData: chartMultiData,
  options: chartOptions,
};

export const ZoomChart = Template.bind({});
ZoomChart.args = {
  chartType: 'bar',
  chartData: chartMultiData,
  options: chartOptions,
  enableZoom: true,
};
