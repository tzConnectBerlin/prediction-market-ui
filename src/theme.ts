import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
    light: 'rgba(212, 241, 204, 0.12)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.04)',
    '300': 'rgba(29, 34, 39, 0.03)',
    '200': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.54)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.54)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
  // action: {
  //   disabledBackground: "#0166FF",
  //   disabled: "rgba(255,255,255,0.3)",
  // },
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
});

export const darkTheme = createTheme({
  palette: {
    ...lightThemePalette,
    primary: {
      dark: 'rgba(214, 231, 255, 1)',
      main: 'rgba(204, 224, 255, 1)',
      light: 'rrgba(204, 224, 255, .12)',
      contrastText: 'rgba(29, 34, 39, 0.87)',
    },
    secondary: {
      dark: 'rgba(204, 224, 255, 0.14)',
      main: 'rgba(204, 224, 255, 0.12)',
      light: 'rgba(204, 224, 255, 0.14)',
      contrastText: 'rgba(204, 224, 255, 1)',
    },
    tertiary: {
      dark: 'rgba(212, 241, 204, 1)',
      main: 'rgba(212, 241, 204, 1)',
      light: 'rgba(212, 241, 204, 0.12)',
    },
    grey: {
      '900': 'rgba(255, 255, 255, 1)',
      '800': 'rgba(255, 255, 255, 0.87)',
      '700': 'rgba(255, 255, 255, 0.38)',
      '600': 'rgba(255, 255, 255, 0.2)',
      '500': 'rgba(255, 255, 255, 0.11)',
      '400': 'rgba(255, 255, 255, 0.09)',
      '300': 'rgba(255, 255, 255, 0.03)',
      '200': 'rgba(255, 255, 255, 0.02)',
    },
    text: {
      primary: 'rgba(255, 255, 255, 0.87)',
      secondary: 'rgba(255, 255, 255, 0.54)',
      disabled: 'rgba(255, 255, 255, 0.38)',
    },
    buttonText: {
      primary: 'rgba(29, 34, 39, 0.87)',
      secondary: 'rgba(29, 34, 39, 0.54)',
      disabled: 'rgba(29, 34, 39, 0.38)',
    },
    background: {
      default: 'rgba(29, 34, 39, 1)',
      paper: 'rgba(45, 49, 54, 1)',
    },
    mode: 'dark',
  },
  zIndex: {
    tooltip: 9999,
  },
});
