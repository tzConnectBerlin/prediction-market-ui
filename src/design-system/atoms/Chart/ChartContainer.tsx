import React from 'react';
import styled from '@emotion/styled';
import { Paper } from '@material-ui/core';
import { Typography } from '../Typography';
import { Chart, ChartProps } from './Chart';

const PaperWrapperStyled = styled(Paper)`
  padding: 2rem;
`;

export const ChartContainer: React.FC<ChartProps> = ({ chartTitle, ...props }) => {
  return (
    <PaperWrapperStyled>
      <Typography size="h5" fontWeight="bold">
        {chartTitle}
      </Typography>
      <Chart {...props} />
    </PaperWrapperStyled>
  );
};
