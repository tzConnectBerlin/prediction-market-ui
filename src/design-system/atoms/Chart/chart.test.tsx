import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Chart } from './Chart';
import { lightTheme as theme } from '../../../theme';

const defaultData = {
  labels: ['3 MAR 2021', '4 MAR 2021', '5 MAR 2021', '6 MAR 2021', '7 MAR 2021'],
  datasets: [
    {
      label: 'Yes',
      data: [1, 2, 3, 2.5, 4],
      borderColor: theme.palette.success.main,
      backgroundColor: theme.palette.success.main,
      yAxisID: 'y',
    },
  ],
};

describe('Snapshot testing Chart Component', () => {
  it('renders correctly Line chart', () => {
    const ChartEl = renderer.create(<Chart chartData={defaultData} />).toJSON();
    expect(ChartEl).toMatchSnapshot();
  });

  it('renders correctly Bar chart', () => {
    const ChartEl = renderer.create(<Chart chartData={defaultData} chartType="bar" />).toJSON();
    expect(ChartEl).toMatchSnapshot();
  });

  it('renders correctly title', () => {
    const ChartEl = renderer
      .create(<Chart chartData={defaultData} chartTitle="Test Title" />)
      .toJSON();
    expect(ChartEl).toMatchSnapshot();
  });
});
