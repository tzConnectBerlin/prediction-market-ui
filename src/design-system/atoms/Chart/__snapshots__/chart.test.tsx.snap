// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Chart Component renders correctly Bar chart 1`] = `
<canvas
  data-testid="canvas"
  height={150}
  onClick={[Function]}
  role="img"
  typeof="bar"
  width={300}
/>
`;

exports[`Snapshot testing Chart Component renders correctly Line chart 1`] = `
<canvas
  data-testid="canvas"
  height={150}
  onClick={[Function]}
  role="img"
  typeof="line"
  width={300}
/>
`;

exports[`Snapshot testing Chart Component renders correctly title 1`] = `
<canvas
  data-testid="canvas"
  height={150}
  onClick={[Function]}
  role="img"
  typeof="line"
  width={300}
/>
`;
